sciris/weblib/scirisobjects.py

```diff
@@ -186,53 +186,96 @@ class ScirisCollection(ScirisObject):
                     
     Attributes:
         obj_dict (dict) -- the Python dictionary holding the ScirisObjects
+        objs_within_coll (bool) -- are the objects themselves within the 
+            collection? If they are not, they are kept as separate data store 
+            entries and handle_dict is used.
+        ds_uuid_set (set) -- the Python set holding UUIDs to DataStore 
+            entries for the ScirisObjects (used only if objs_within_coll is False)
         
     Usage:
         >>> objs = ScirisCollection(uuid.UUID('12345678123456781234567812345678'))                      
     """
     
     def __init__(self, uid, type_prefix='collection', 
-        file_suffix='.scl', instance_label=''):
+        file_suffix='.scl', instance_label='', objs_within_coll=False):
         # Set superclass parameters.
         super(ScirisCollection, self).__init__(uid, type_prefix, 
              file_suffix, instance_label)
         
         # Create a Python dict to hold the ScirisObjects.
         self.obj_dict = {}
         
+        # Set the flag for whether the objects are stored within the collection 
+        # itself (or outside in separate data store entries).
+        self.objs_within_coll = objs_within_coll
+        
+        # Set the datastore UUID set to empty.
+        self.ds_uuid_set = set()
+        
     def load_from_copy(self, other_obj):
         if type(other_obj) == type(self):
             # Do the superclass copying.
             super(ScirisCollection, self).load_from_copy(other_obj)
             
+            # Copy other items specific to this class.
             self.obj_dict = other_obj.obj_dict
+            self.objs_within_coll = other_obj.objs_within_coll
+            self.ds_uuid_set = other_obj.ds_uuid_set
             
     def get_object_by_uid(self, uid):
         # Make sure the argument is a valid UUID, converting a hex text to a
         # UUID object, if needed.
         valid_uid = ut.uuid(uid)
         
-        # If we have a valid UUID, return the matching ScirisObject (if any); 
-        # otherwise, return None.
+        # If we have a valid UUID...
         if valid_uid is not None:
-            return self.obj_dict.get(valid_uid, None)
+            # If we have the objects within the collection itself...
+            if self.objs_within_coll:
+                # Return the object.
+                return self.obj_dict.get(valid_uid, None)
+            
+            # Otherwise...
+            else:
```

Comment:
Probably should check that the UID is in self.ds_uuid_set here, otherwise it will return objects even if they aren't contained within this collection

Reply:
Good catch.  I just fixed that, and will push.